tests/utils/async.nim

```diff
@@ -0,0 +1,30 @@
+import chronos/futures, stew/results, chronos
+import ./futures
+
+proc toOk(future: Future[void]): Result[void, string] =
+  return results.ok()
+
+proc toOk[T](future: Future[T]): Result[T, string] =
+  return results.ok(future.read())
+
+proc toResult*[T](future: Future[T]): Result[T, string] =
+  if future.cancelled():
+    return results.err("Future cancelled/timed out.")
+  elif future.finished():
+    if not future.failed():
+      return future.toOk()
+    else:
+      return results.err("Future finished but failed.")
+  else:
+    return results.err("Future still not finished.")
+
+proc waitForResult*[T](
+    future: Future[T], timeout = DURATION_TIMEOUT
+): Future[Result[T, string]] {.async.} =
+  discard await future.withTimeout(timeout)
+  return future.toResult()
+
+proc reset*[T](future: Future[T]): void =
```

Comment:
We shouldn't access internal fields.

Reply:
This function is to be used only during testing, thus being inside `tests/utils/async.nim`. It's a not-perfect-but-good-enough shortcut to reuse futures in handlers, for instance.

Reply:
As I previously stated, it's just a test-only workaround. Regardless, I'm happy to delete it and add it back when I need it.

Reply:
We shouldn't be introducing workarounds in the codebase unless they are strictly necessary. Is there a `Future` feature that is missing and this workaround is fixing?

Reply:
Just a handier way of resetting a future.  During testing they need to be reset one way or another, even if it doesn't see, that way at first.

But don't worry, I'll happily remove until I make clear use of it.